### DISCUSSION TITLE
How to replace a groupped object with another?

### DISCUSSION THREAD

**Discussion by enriqueerrando** [fabricjs/fabric.js#8480] [Q&A] [answered]
Hi,

Let's say I have a group with a circle and a square. I want to add a triangle that replaces the circle. 

I've been playing around, but connot manage to remove the circle and addWithUpdate the triangle, for some reason the triangle doesn't get added to the group. So my final approach was to manually ungroup and regroup, but I don't really like this solution.

Thanks in advance!

**Comment by iocoker** [comment]
@enriqueerrando Do you have a sample code?

**Comment by ShaMan123** [comment, marked as answer]
It is good to see that many devs will benefit from #7670 
If I were you I would workaround with as less effort as possible (ungroup, regroup sounds good) and wait for v6.

**Reply by ShaMan123** [reply]
It will not be long since we release
Expected by 1/1/23

**Reply by ShaMan123** [reply]
In v6 you would do something like:

Using **relative** position as measured by group
```ts
const circle = group.getObjects('circle')[0];
const relativePosition = circle.getRelativeCenterPoint();
group.remove(circle);
group.add(triangle);
triangle.setRelativeXY(relativePosition, 'center', 'center');
```

Using position as measured by canvas
```ts
const circle = group.getObjects('circle')[0];
const position = circle.getCenterPoint();
triangle.setXY(position, 'center', 'center');
group.remove(circle);
group.add(triangle);


```
The benefit of the second approach is that it doesn't matter when we measure the center point or when we call `setXY`

**Reply by enriqueerrando** [reply]
Thanks @ShaMan123 it's great that we will be able to work with groups as we do with canvas!!

I agree, since the release date is just around the corner and will include a fix for this, make sense to just wait for it and meanwhile continue using the ungroup-regroup solution ;)
